controllers/clusterversion_controller.go

```diff
@@ -134,6 +140,14 @@ func (c *ClusterVersionReconciler) Reconcile(ctx context.Context, req ctrl.Reque
 		return ctrl.Result{}, err
 	}
 
+	if deployCSI, err := c.getDeployCSIConfig(); err != nil {
+		c.log.Error(err, "failed to perform precheck for deploying CSI")
+		return ctrl.Result{}, err
+	} else if !deployCSI {
+		c.log.Info("not deploying CSI as not indicated in the operator config map")
+		return ctrl.Result{}, nil
+	}
```

Comment:
Don't assume that all of the code below this check is CSI reconciliation code. Let's reverse the condition and nest the CSI reconciliation code. You can even move it into its own reconcileCSI func to make it easier to follow

Reply:
done.